cpp/src/arrow/engine/substrait/options.h

```diff
@@ -0,0 +1,55 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// This API is EXPERIMENTAL.
+
+#pragma once
+
+namespace arrow {
+namespace engine {
+
+/// How strictly to adhere to the input structure when converting between Substrait and
+/// Acero representations of a plan. This allows the user to trade conversion accuracy
+/// for performance and lenience.
+enum class ConversionStrictness {
+  /// Prevent information loss by rejecting incoming plans that use features or contain
+  /// metadata that cannot be exactly represented in the output format in a way that
+  /// will round-trip. Relations/nodes must map one-to-one.
+  PEDANTIC,
```

Comment:
"Pedantic" conveys an idea of spurious or arbitrary constraints. Is this the case here? Otherwise "strict" would sound better IMHO.

Reply:
> "Pedantic" conveys an idea of spurious or arbitrary constraints. Is this the case here?

I'd say so, actually. "Pedantic" includes constraints such as rejecting an Acero plan for conversion to Substrait if the internal column names are not exactly what you'd get after round-tripping through Substrait (which cannot represent inner column names), even though the plan can be conveyed in Substrait perfectly well.

> Otherwise "strict" would sound better IMHO.

Strict implies to me to strictly conform to all the rules of the output format, i.e. don't make anything that's out of spec. But none of the options do that, so it feels kind of meaningless.

Reply:
Hmm... perhaps "exact_roundtrip" or "perfect_roundtrip" based on what I'm reading below?

Reply:
Sure, that probably captures it better.